### PR TITLE
chore: add transformers

### PR DESCRIPTION
**Pull request by KFlash**
This implements an transformer API and a visitor to transform the `RootFile`.

It will transform and get rid of Kataw's own types, but can be extended and new transformers can be added.

TODO:
-------

- [x]  Traverse and update all CST nodes on changes

- [x]  Handle lexical scoping

- [x]  Allow swapping of keywords

- [x]  Get rid of all Kataw types

- [ ]  Performance

### CODE REVIEW

**Comment by KFlash** [comment]
@aladdin-add The goal is to replace current walker with the one in his PR. Can you find a way to test the transformers?

**Comment by KFlash** [comment]
@aladdin-add Removing Kataw types now works. Can you test this?  I'm going to finish up the transformer now and focus on it's performance

File: src/transform/visitor.ts [hunk 1]
```diff
@@ -0,0 +1,1259 @@
...
+  LexicalEnvironmentFlags,
+  createNodeArray,
+  extractSingleNode,
+  startLexicalEnvironment,
+  endLexicalEnvironment,
+  concatenate
+} from './core';
+
+export function visitEachChild(transform: Transform, node: any, visitor: (node: SyntaxKind) => SyntaxKind): any {
+  const kind = node.kind;
```
Comment (aladdin-add) [inline src/transform/visitor.ts:140]: have you tested it? the type node: SyntaxKind seems not correct.

Comment (KFlash) [reply src/transform/visitor.ts:140]: I'm working on testing and improve it locale. There are bugs that I fixed already, but not pushed

**Comment by KFlash** [comment]
@aladdin-add Can you add this to the test runner?

File: src/transform/visitor.ts (hunk 1 above)
Comment (KFlash) [reply src/transform/visitor.ts:140]: The SyntaxKind is correct, you need to do a type cast to get it correct. Se my update

**Comment by aladdin-add** [comment]
> Can you add this to the test runner?

not sure what you mean by "add this to the test runner". can you outline what you want to test, and what the API looks like?

**Comment by aladdin-add** [comment]
maybe off-topic: we cannot test all the things in the snapshots - we will have to introduce something like mocha/jest/node-tap to writing some unit tests.

**Comment by KFlash** [comment]
@aladdin-add You remove the current walker and use the new on instead in the test runner - visitor.ts. Then just walk !

**Comment by KFlash** [comment]
@aladdin-add then add back mocha

**Comment by KFlash** [comment]
@aladdin-add The visitor should be okay. I fixed the types and corrected a bunch of bugs. Can you replace the old walker?

And I think we merge this as is and do follow up PR for lexicals and other transformers

**Comment by KFlash** [comment]
@aladdin-add I added a 'removeKatawTypes' as public API. See the readme. It need to be tested in the test runner

I'm merging this now

**Pull request merged**